### PR TITLE
small improvements to Morphism.Reasoning

### PR DESCRIPTION
**Pull request by JacquesCarette**
Add more documentation to Categories.Morphism.Reasoning.Core.  Make some module names more reasonable. Simplify some proofs. Fix up downstream code that broke.

### CODE REVIEW

**Review by TOTBWF — APPROVED** [review]
Adding explicit parens to the type signatures is a good move!

File: src/Categories/Morphism/Reasoning/Core.agda [hunk 1]
```diff
@@ -5,11 +5,16 @@ open import Categories.Category
   Helper routines most often used in reasoning with commutative squares,
   at the level of arrows in categories.
 
-  Basic  : reasoning about identity
+  Identity : reasoning about identity
+  Assoc4   : associativity short-cuts for 4-term
```
Comment (Taneb) [inline src/Categories/Morphism/Reasoning/Core.agda:9]: I'm not sure this line makes grammatical sense

**Review by Taneb — APPROVED** [review]
This is definitely a great improvement

File: src/Categories/Morphism/Reasoning/Core.agda (hunk 1 above)
Comment (JacquesCarette) [reply src/Categories/Morphism/Reasoning/Core.agda:9]: You are right -- will fix.

**Comment by JacquesCarette** [comment]
@TOTBWF I have no idea what mechanism produces these "dismissed stale review", I find them rather rude, and it wasn't my explicit doing!

**Pull request merged**